### ISSUE TITLE
ENH: gradient clipping

### ISSUE THREAD

**Issue by Xylambda** [Xylambda/torchfitter#24]
Add [gradient clipping](https://stackoverflow.com/questions/54716377/how-to-do-gradient-clipping-in-pytorch) to the training loop.
